app/policies/article_policy.rb

```diff
@@ -0,0 +1,45 @@
+class ArticlePolicy < ApplicationPolicy
+  authorize :user, allow_nil: true
+
+  relation_scope do |relation|
+    if user.nil?
+      relation.published
+    else
```

Comment:
Maybe it will be better to swap if-else?
```suggestion
    if user.present?
    ...
```

Reply:
good, it looks prettier